ansible/verify_config.py

```diff
@@ -0,0 +1,768 @@
+                is_valid = False
+
+        return is_valid
+
+
+class HostNetworkValidation(Validator):
+    def __init__(self):
+        super().__init__()
+
+    def validate(self):
+        self.assertion.assert_true(
+            lambda: self._check_if_bridge_is_up(),
+            reason=f"Interface 'br1' is not up. Consider running "
+                   f"'{Formatting.yellow('./setup-management-network.sh')}'",
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_proxy_is_using(),
+            reason="Proxy setting is not correct ",
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_docker_registry(),
+            reason="Cannot establish a connection to docker registry. Check your proxy setting, docker proxy setting"
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_apt_repository(),
+            reason="Cannot establish a connection to apt repository. Check your proxy setting."
+        )
+
+    def _check_if_can_connect_to_apt_repository(self):
+        return_code = Utility.execute_shell("apt-get --simulate upgrade")
+        return return_code == 0
+
+    def _check_if_bridge_is_up(self):
+        return_code = Utility.execute_shell("ifconfig br1")
+        return return_code == 0
+
+    def _check_if_can_connect_to_docker_registry(self):
+        is_docker_installed = Utility.execute_shell("command docker -v", is_shell=True) == 0
+
+        if not is_docker_installed:
+            self.assertion.add_error_details("Docker is not installed on the system. Please install docker")
+            return False
+
+        docker_image_url = f"{Config.DOCKER_REGISTRY_URL}/{Config.DOCKER_IMAGE_NAME}:latest"
+
+        # We need to check for pull here since only pull use the setting from docker-proxy. Unless in the future
+        # there are other alternatives. If fail exit code will be `1` otherwise will be `124` exit code for `timeout`
+        can_pull_image = Utility.execute_shell(f"timeout 1s docker pull {docker_image_url}") == 124
+
+        if not can_pull_image:
+            self.assertion.add_error_details("Not able to connect to docker-registry. Please configure your proxy")
+            return False
+
+        return True
+
+    def _check_if_proxy_is_using(self):
+        group_vars_env = Utility.parse_yml(Config.GROUP_VARS_ENV_FILE)
+
+        env_vars = ["http_proxy", "https_proxy"]
+
+        for var in env_vars:
+            if var not in os.environ:
+                continue
+
+            if var not in group_vars_env or var not in group_vars_env['proxy_env']:
+                self.assertion.add_error_details(
+                    f"'{Formatting.red(var)}' is detected in environment variables "
+                    f"but not in '{Config.GROUP_VARS_ENV_FILE}'",
+                )
+                return False
+
+            if group_vars_env[var] != os.environ[var]:
```

Comment:
I think `group_vars_env` is a dictionary with `proxy_env` as the key, right? `http_proxy` and `https_proxy` is one level under.

Reply:
Thanks for pointing that out. That's correct.